### ISSUE TITLE
Error when branch moves on after build starts

### ISSUE THREAD

**Issue by JakeGinnivan** [GitTools/GitVersion#628]
```
WARN [09/02/15 23:46:29:27] An error occurred:
Couldn't find any remote tips from remote 'https://github.com/GitTools/GitVersion.git' pointing at the commit '7d833b736e37f07aa9d020efb363e72b2af7dcb4'.
Command exited with code 1
```

This happens when the build kicks off then the branch you are building moves on. Need to investigate if this can be fixed.

**Comment by JakeGinnivan** [comment]
This will be fixed on AppVeyor, TFS and Jenkins with #755

Cannot fix teamcity OOTB.

**Comment by JakeGinnivan** [comment]
Also made improvements to TeamCity docs and introduced Git_Branch to get this info from TeamCity
